helix-loader/build.rs

```diff
@@ -23,4 +24,40 @@ fn main() {
 
     println!("cargo:rerun-if-changed=../VERSION");
     println!("cargo:rustc-env=VERSION_AND_GIT_HASH={}", version);
+
+    if git_hash.is_none() {
+        return;
+    }
+
+    // we need to revparse because the git dir could be anywhere if you are
+    // using detached worktrees but there is no good way to obtain an OsString
+    // from command output so for now we can't accept non-utf8 paths here
+    // probably rare enouch where it doesn't matter tough we could use gitoxide
+    // here but that would be make it a hard dependency and slow compile times
+    let Some(git_dir): Option<String> = Command::new("git")
+        .args(["rev-parse", "--git-dir"])
+        .output()
+        .ok()
+        .filter(|output| output.status.success())
+        .and_then(|x| String::from_utf8(x.stdout).ok())
+    else{ return; };
```

Comment:
Does `cargo fmt` work on build scripts?

Reply:
yes but rustfmt currently ignores `let .. else` (leaves it unchanged) altough that will hopefully be fixed in the not-so-distant future